### PR TITLE
Move failing tests to unstable #1475

### LINKED ISSUE #1475: Move failing tests to unstable
Moves tests that fail consistently to unstable folder

### PR DESCRIPTION
**Pull request by jinified**
Fixes #1475 

## Observation
1. Tests seem to pass fine at night when there are less parallel build jobs.
2. NPE when clicking because node not part of a scene

## Failing frequently
1. UIEventTest.panelClick
2. FilterTextFieldTest

## Infrequent
1. LabelPicker, RepositoryPicker
2. UIComponentTest

## Tips
- favors triggering event directly if possible rather than using UI control such as clicking 
- use `traverseMenu()` over multiple `clickOn`
- try to `waitUntilNodeAppears` before continuing with the rest of logic
- use `waitAndAsserts` over `assertEquals` for operation that may take some time

### CODE REVIEW

**Comment by jinified** [comment]
Update: Manages to pass 10 times consecutively in the evening till late night.

**Comment by damithc** [comment]
@jinified consider spawning off multiple branches and trying different fixes in parallel. But don't overdo it because it will block other PRs from running tests.

**Comment by jinified** [comment]
I think this should be more stable for now compared to the current version we have on master

**Comment by damithc** [comment]
Let us know when ready to review.

**Comment by jinified** [comment]
@dariusf  Finally ready for review. Some of the solutions are temporary such as sleeping before certain action. These will be fixed in the other PR.

File: .travis.yml [hunk 1]
```diff
@@ -6,13 +6,13 @@ jdk:
...
+#    - rm -f $HOME/.gradle/caches/modules-2/modules-2.lock
 
-cache:
-  directories:
-    - $HOME/.gradle/caches/
-    - $HOME/.gradle/wrapper/
+#cache:
+#  directories:
+#    - $HOME/.gradle/caches/
+#    - $HOME/.gradle/wrapper/
```
Comment (damithc) [inline .travis.yml]: Remove commented code?

Comment (jinified) [inline .travis.yml]: Noted

**Comment by jinified** [comment]
Changes were made . Ready for review

**Comment by damithc** [comment]
Looks like this PR passed the last few times. Let's merge this soon. @dariusf or @ndt93 can review?

**Comment by ndt93** [comment]
No comments from me for this. @dariusf Any comment?

File: src/test/java/guitests/UITest.java [hunk 2]
```diff
@@ -387,15 +387,17 @@ public FlowPane getAssigneePickerAssignedUserPane() {
      * @param panelIndex
      */
     public void clickFilterTextFieldAtPanel(int panelIndex) {
-        waitUntilNodeAppears(IdGenerator.getPanelFilterTextFieldIdReference(panelIndex));
-        clickOn(IdGenerator.getPanelFilterTextFieldIdReference(panelIndex));
+        sleep(EVENT_DELAY);
```
Comment (dariusf) [inline src/test/java/guitests/UITest.java]: Maybe we can put the reason for this in the comment?

**Comment by dariusf** [comment]
Apart from that, looks good to me.

**Comment by jinified** [comment]
Changes were made.

File: src/test/java/guitests/UITest.java [hunk 3]
```diff
@@ -74,7 +74,7 @@
      * as suggested by TestFX documentation
      */
     static {
-        if (Boolean.getBoolean("headless")) {
+        if (true || Boolean.getBoolean("headless")) {
```
Comment (damithc) [inline src/test/java/guitests/UITest.java]: Intended?

Comment (jinified) [inline src/test/java/guitests/UITest.java]: Not intended. But need to remove ContextMenuTest after updating from master.

**Comment by jinified** [comment]
@damithc  Are you ok with the current state of the code excluding my blunder that you mentioned ? I will make sure to run a few more tests to ensure updating master does not make the tests unstable

**Comment by damithc** [comment]
Yes, can merge after that.

**Pull request merged**